plugin.rb

```diff
@@ -52,61 +56,37 @@ def after_authenticate(auth_token)
         end
       end
     end
-    if trustedGuild && !User.find_by_email(auth_token.info.email)
-      systemUser = User.find_by(id: -1)
-      Invite.generate_invite_link(auth_token.info.email, systemUser)
-    end
 
     result = super
-    data = auth_token[:info]
-    result.extra_data[:avatar_url] = data[:image]
-    if (avatar_url = data[:image]).present?
-      retrieve_avatar(result.user, avatar_url)
-    end
+
     if trustedGuild
       result.extra_data[:auto_approve] = true
     else
       result.extra_data[:auto_approve] = false
     end
+
     result
   end
 
   def after_create_account(user, auth)
     super
+
     data = auth[:extra_data]
     if !user.approved && data[:auto_approve]
-      user.approve(-1, false)
-    end
-    if (avatar_url = data[:avatar_url]).present?
-      retrieve_avatar(user, avatar_url)
+      user.approved = true
+      user.approved_by_id = Discourse.system_user.id
+      user.save!
+      if reviewable = ::ReviewableUser.pending.find_by(target: user)
+        reviewable.perform(:approve, Discourse.system_user)
```

Comment:
This `ReviewableUser` code is never hit, because the reviewable object is created after `after_create_account` is run. That's why the manual `user.approved = true` is required to make it work, and why there is no email notification sent to users.

https://github.com/discourse/discourse/blob/f7809207591c6ec48d137df9b7fc05514aa80dd5/app/controllers/users_controller.rb#L414-L422

We might be able to work around this, but I wonder if we should actually use change the approach here. I posted [on meta about it](https://meta.discourse.org/t/discord-oauth2-plugin/67174/59?u=david)